core/src/filed/backup.cc

```diff
@@ -1188,15 +1188,15 @@ static int send_data(JobControlRecord* jcr,
     /*
      * To read FreeBSD partitions, the read size must be a multiple of 512.
      */
-    bctx.rsize = (bctx.rsize / 512) * 512;
+    bctx.rsize = ((bctx.rsize + 512) / 512) * 512;
```

Comment:
I guess `bctx.rsize` is the size of `bctx.rbuf`. Is that buffer allocated with enough space?

Reply:
As far as I understand the code the previous version rounded down so rsize is a multiple of 512 and does not exceed its original size.

Reply:
I understand i.e.: 
(rsize=511 / 512) * 512 = 0  -> 511 Bytes too few
(rsize=513 / 512) * 512 = 512 -> 1 Byte too few

So I should check the allocation of bctx.rbuf, not round down to a smaller value.

Reply:
Now I understood from code-review, that bctx.rsize will be initialized by the maximum buffer length (that is multiple of 512) and rounded down to multiple of 512. The case with the result 0, therefore, won't happen.
bctx.rsize in turn is used to limit the copied amount of data into the buffer bctx.rbuf.